common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -122,6 +131,37 @@ protected void append(LoggingEvent event) {
             System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
             isTruststore = true;
         }
+        if (dataPublisher == null) {
```

Comment:
Hi Ruwan,

This is fixed by the commit in https://github.com/wso2/shared-analytics/commit/eb6f50ab69a03c7b98f0a9aca51cc43b9139b46e

Thanks
SajithD